smartsim/_core/launcher_/slurm/slurm_parser.py

```diff
@@ -121,7 +121,10 @@ def parse_sstat_nodes(output: str, job_id: str) -> t.List[str]:
     return list(set(nodes))
 
 
-def parse_step_id_from_sacct(output: str, step_name: str) -> t.Optional[str]:
+StepID = t.NewType("StepID", str)
```

Comment:
I would _LOVE_ to, but seeing as we are kinda shifting focus away from using launchers directly and more toward dragon, I wasn't planning on tackling it immediately. The `NewType` here is just so that the type checker could help me keep my sanity, lol.

I could definitely wrap the two tickets previously requested into a "revamp slurm utilities" style ticket for when we have more time to work on this!!